### ISSUE TITLE
History navigation with arrow up/down throws me off

### ISSUE THREAD

**Issue by jooize** [vector-im/riot-web#5440]
I get confused and annoyed when navigating my message and inadvertently end up on a previous message. I'm the worst. Change the keys. 💯

**Comment by lampholder** [comment]
What would you prefer? I make use of this feature regularly, so I wouldn't like to see it removed completely.

<kbd>&uarr;</kbd>/<kbd>&darr;</kbd> taking you backwards and forwards through the history of your own per-channel messages is pretty common on IRC clients.

Slack and Gitter take you to an "edit just the last message you just typed" box if you press <kbd>&uarr;</kbd> beyond the bounds of your current message.

WhatsApp and Messenger don't support this feature.

> Change the keys.

On a personal note, I triage a lot of issues, and nearly everyone is polite, enthusiastic and a genuine pleasure to engage with. I find it a little upsetting, then, to see a request issued as a blunt instruction. Requests are welcome, suggestions are awesome; instructions come across as rude :disappointed:

**Comment by jooize** [comment]
> On a personal note, I triage a lot of issues, and nearly everyone is polite, enthusiastic and a genuine pleasure to engage with. I find it a little upsetting, then, to see a request issued as a blunt instruction. Requests are welcome, suggestions are awesome; instructions come across as rude 😞

Yes, I'm sorry. Occasonally, I naïvely hope that I can pull something like this and people will just understand that it's meant to be mildly amusing.

I will consider alternatives! Usually, I'd have some to give, but I wanted to get this conversation started somehow.

**Comment by lampholder** [comment]
Heh, fair enough :) That's the problem with text - tone can easily get lost in translation :grinning: 

One could make a case for the message composer's being the most important part of the UX, so all feedback _is_ welcome - hopefullly we can contrive something which meets most people's needs out the box and provides enough customisation that everybody can have the experience they want (without needing a space shuttle dashboard of configuration options :stuck_out_tongue:)

**Comment by jooize** [comment]
I agree with those goals. It’s somewhat unexpected I have a problem with this as I’m comfortable in WeeChat or Irssi, and actually want arrow down to push the message into the arrow up history, but it’s only when the composer is multi-line I’m disoriented. Happened when trying multi-line in WeeChat too.

Might be that I don’t know a more convenient reliable way of reaching the top and start of line except arrow up and therefore eagerly press it too many times.

I hope we can preserve most native macOS keyboard actions.

**Comment by jooize** [comment]
⌘↑ is what I should be using to get to the start of the first line. Wish Apple hadn't ruined the arrow keys on new MacBooks. In a year of daily use I haven't learned to quickly feel whether my little finger is on ⇧, ↑, or ↓.

By the way, should I have placed this issue in another repository?

**Comment by aaronraimist** [comment]
Maybe this is somewhat better now that the arrow keys edit the previous message rather than navigate history

**Comment by yannicka** [comment]
Is the problem still relevant?

**Comment by aaronraimist** [comment]
Imo yes although it is a slightly different issue now. Previously the up/down arrows would make the composer change to show old messages. Now it edits the previous message in line. I still find myself accidentally hitting the up arrow one too many times while trying to send a message and ending up accidentally editing my previous message.

I think the solution should be to turn off the up arrow that lets you edit the previous message editing feature when there is text already in the composer.

**Comment by turt2live** [comment]
I think the solution proposed above ( https://github.com/vector-im/element-web/issues/5440#issuecomment-735275744 ) is more than reasonable as a fix.
